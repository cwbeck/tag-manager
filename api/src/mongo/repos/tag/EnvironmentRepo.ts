import Repo from '../../abstractions/Repo';
import { injectable } from 'inversify';
import Environment from '../../models/tag/Environment';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class EnvironmentRepo extends Repo<Environment> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                _app_id: 1,
            },
        },
    ];
}
