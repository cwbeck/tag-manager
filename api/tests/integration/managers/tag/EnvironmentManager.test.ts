import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import EnvironmentRepo from '../../../../src/mongo/repos/tag/EnvironmentRepo';
import Mock = jest.Mock;

describe('EnvironmentManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a single environment', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();

        const mockSutModel = TestUtils.buildMockEnvironment();
        const mockSutRepo = {} as EnvironmentRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getEnvironment(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
