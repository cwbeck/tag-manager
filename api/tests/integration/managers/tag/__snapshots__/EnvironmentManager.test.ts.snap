// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvironmentManager fetches a single environment 1`] = `
Object {
  "data": Object {
    "getEnvironment": Object {
      "id": "5edf4c48e9615d9a1c66cfff",
      "name": "test environment",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
